### ISSUE TITLE
Service with named provider fails, but Deployment works fine.

### ISSUE THREAD

**Issue by clstokes** [pulumi/pulumi#5573]
### Problem description

If I set a named provider on a `pulumi_kubernetes.core.v1.Service` as shown below, the resource will fail with the error below. If I do the same thing on a `pulumi_kubernetes.apps.v1.Deployment`, it works fine.

I'm not sure if this is a Python-specific problem or affects other k8s resource types as well.

```py
service_with_named_provider=Service(
    "service-with-named-provider",
    spec=my_service_spec,
    opts=pulumi.ResourceOptions(
        depends_on=[deployment_with_named_provider], 
        providers={"kubernetes": k8s_provider}, # DOES NOT WORK
        # providers=[k8s_provider], # ALSO DOES NOT WORK
    ),
)
```

### Errors & Logs

```
Diagnostics:
  kubernetes:core/v1:Service (service-with-named-provider):
    error: configured Kubernetes cluster is unreachable: unable to load Kubernetes client configuration from kubeconfig file: invalid configuration: no configuration has been provided, try setting KUBERNETES_MASTER environment variable
 ```

### Affected product version(s)

```
(venv) k8s-py-nginx % pulumi version 
v2.11.2
(venv) k8s-py-nginx % pip list       
Package           Version
----------------- ---------
Arpeggio          1.9.2
attrs             20.2.0
certifi           2020.6.20
chardet           3.0.4
dill              0.3.2
grpcio            1.32.0
idna              2.10
parver            0.3.1
pip               20.2.3
protobuf          3.13.0
pulumi            2.11.2
pulumi-kubernetes 2.6.2
PyYAML            5.3.1
requests          2.24.0
semver            2.10.2
setuptools        50.3.0
six               1.15.0
urllib3           1.25.10
wheel             0.35.1
```

### Reproducing the issue

Code to reproduce - https://gist.github.com/4c0d5bea3ce19c268b2a9a27fc64fa53

Steps

1. Provision a Kubernetes cluster and save your kubeconfig to `~/Desktop/repro-kubeconfig`.
1. Unset `$KUBECONFIG` and remove `~/.kube/config`.
1. `git clone https://gist.github.com/4c0d5bea3ce19c268b2a9a27fc64fa53.git repro-code`
1. `cd repro-code`
1. `pulumi stack init my-repro`
1. `pulumi up`
1. Observe error

### Suggestions for a fix

Ensure Service can use a named provider like other k8s resources. Also, please check other K8s resources to see if others are affected.

**Comment by EvanBoyle** [comment]
I just had this issue manifest with a helm chart, which is sort of surprising as our chart implementation is a component resource. I guess somewhere the implementation must propagate these options to a custom resource (looks like a service account).

```ts
const argocd = new k8s.helm.v3.Chart("argocd",
    {
        namespace: "default",
        chart: "argo-cd",
        fetchOpts: { repo: "https://argoproj.github.io/argo-helm" },
        values: {
            installCRDs: false,
            configs: {
                secret: {
                    argocdServerAdminPassword: password,
                },
            },
            server: {
                service: {
                    type: 'LoadBalancer',
                },
            }
        },
        // The helm chart is using a deprecated apiVersion,
        // So let's transform it
        // transformations: [
        //     (obj: any) => {
        //         if (obj.apiVersion == "extensions/v1beta1")  {
        //             obj.apiVersion = "networking.k8s.io/v1beta1"
        //         }
        //     },
        // ],
    },
    { providers: { kubernetes: provider }},
);
```

```
    error: Error: Do not supply 'providers' option to a CustomResource. Did you mean 'provider' instead?
        at new CustomResource (/Users/evanboyle/go/src/eksPanic/node_modules/@pulumi/pulumi/resource.js:301:19)
        at new ServiceAccount (/Users/evanboyle/go/src/eksPanic/node_modules/@pulumi/core/v1/serviceAccount.ts:95:9)
        at /Users/evanboyle/go/src/eksPanic/node_modules/@pulumi/yaml/yaml.ts:3136:27
        at /Users/evanboyle/go/src/eksPanic/node_modules/@pulumi/pulumi/output.js:249:35
        at Generator.next (<anonymous>)
        at /Users/evanboyle/go/src/eksPanic/node_modules/@pulumi/pulumi/output.js:21:71
        at new Promise (<anonymous>)
        at __awaiter (/Users/evanboyle/go/src/eksPanic/node_modules/@pulumi/pulumi/output.js:17:12)
        at applyHelperAsync (/Users/evanboyle/go/src/eksPanic/node_modules/@pulumi/pulumi/output.js:228:12)
        at /Users/evanboyle/go/src/eksPanic/node_modules/@pulumi/pulumi/output.js:182:65

```

@justinvp is this case covered by your suggestions above, or should I file a separate issue?

**Comment by mikhailshilkov** [comment]
I believe this is closed in #8882, see #8796 for details. This hasn't been released yet but will come in 3.25.0.
